### PR TITLE
mgr/dashboard: fix broken alert generator

### PR DESCRIPTION
**Pull request by nizamial09**
Currently the alert generator is broken if you try to run `tox -ealerts-fix`. I fixed it and ran the command and it built a new json file as well.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: monitoring/ceph-mixin/README.md [hunk 1]
```diff
@@ -73,4 +73,10 @@ The jsonnet code located in this directory depends on some Jsonnet third party
 libraries. To update those libraries you can run `jb update` and then update
 the generated files using `tox -egrafonnet-fix`.
 
+### Building alerts from prometheus_alerts.libsonnet
+
+To rebuild the prometheus_alerts.yml file from the corresponding libsonnet,
```
Comment (anthonyeleven) [inline monitoring/ceph-mixin/README.md:78]: Suggest backticks around `prometheus_alerts.yml`, and `libsonnet template`

**Review by anthonyeleven — CHANGES_REQUESTED** [review]

File: monitoring/ceph-mixin/README.md (hunk 1 above)
Comment (nizamial09) [reply monitoring/ceph-mixin/README.md]: done, thanks!

File: monitoring/ceph-mixin/alerts.jsonnet [hunk 2]
```diff
@@ -1 +1 @@
-std.manifestYamlDoc((import 'alerts.libsonnet').prometheusAlerts, indent_array_in_object=true, quote_keys=false)
+std.manifestYamlDoc((import 'mixin.libsonnet').prometheusAlerts)
```
Comment (epuertat) [inline monitoring/ceph-mixin/alerts.jsonnet:1]:
```suggestion
std.manifestYamlDoc((import 'mixin.libsonnet').prometheusAlerts, indent_array_in_object=true, quote_keys=false)
```

**Review by epuertat — DISMISSED** [review]

**Comment by guits** [comment]
@nizamial09 I cherry-picked this commit but this is still failing for me:
```
alerts-fix run-test: commands[0] | jsonnet -J vendor -S alerts.jsonnet -o prometheus_alerts.yml
RUNTIME ERROR: function has no parameter quote_keys
        alerts.jsonnet:1:1-112  
ERROR: InvocationError for command /usr/bin/jsonnet -J vendor -S alerts.jsonnet -o prometheus_alerts.yml (exited with code 1)
```

**Comment by nizamial09** [comment]
> @nizamial09 I cherry-picked this commit but this is still failing for me:
> 
> ```
> alerts-fix run-test: commands[0] | jsonnet -J vendor -S alerts.jsonnet -o prometheus_alerts.yml
> RUNTIME ERROR: function has no parameter quote_keys
>         alerts.jsonnet:1:1-112  
> ERROR: InvocationError for command /usr/bin/jsonnet -J vendor -S alerts.jsonnet -o prometheus_alerts.yml (exited with code 1)
> ```

@guits it'd be because of the `jsonnet` version. check if the version is >=18. I also had this issue with the default package in fedora 37. I had to start a f38 container and run the script there!

**Comment by cloudbehl** [comment]
@nizamial09 should be update the doc for mixin to say that we need jsonnet 18 or above.

**Comment by nizamial09** [comment]
> @nizamial09 should be update the doc for mixin to say that we need jsonnet 18 or above.

@cloudbehl it was already there: https://github.com/ceph/ceph/blob/954e18d2cff388076a954c75d577dd0ffaed6e87/monitoring/ceph-mixin/README.md#any-upgrade-or-downgrade-to-different-major-versions-of-the-recommended-tools-mentioned-above-is-not-supported

**Pull request merged**